include/cute/atom/copy_traits_xe.hpp

```diff
@@ -122,9 +122,10 @@ CUTE_HOST_DEVICE auto prefetch_selector(Tensor const& tensor) {
   #define RETURN_STATEMENT(NON_CONTIG, DTYPE_SIZE, CONTIG) \
     using PrefetchTraits = Copy_Traits<XE_2D_U##DTYPE_SIZE##x##NON_CONTIG##x##CONTIG##_LD_N, decltype(tensor.stride())>; \
     using PrefetchAtom = Copy_Atom<PrefetchTraits, dtype>; \
-    constexpr auto scalar = cute::max(1, DTYPE_SIZE / dtype_size_bits); \
-    using ScalarPrefetchShape = Shape<Int<size<0>(typename PrefetchTraits::BlockShape{})>, \
-                                      Int<size<1>(typename PrefetchTraits::BlockShape{}) * scalar>>; \
+    using Scalar = Int<cute::max(1, DTYPE_SIZE / dtype_size_bits)>; \
+    using ScalarLayout = std::conditional_t<is_need_reversed, Layout<Shape<Scalar, _1>>, Layout<Shape<_1, Scalar>>>; \
+    using ScalarPrefetchShape =  decltype(product_each(raked_product(ScalarLayout{}, \
+                                                        Layout<typename PrefetchTraits::BlockShape>{}).shape())); \
```

Comment:
@t4c1 could you check this? I am less familiar with the prefetch logic & ongoing changes. :pray:

Reply:
That looks fine.